File: python_backend/app/application/use_cases/set_default_persona.py
"""Promote a persona to the user's default."""
from __future__ import annotations

from app.application.ports.personas_repository import PersonasRepository
from app.domain.entities.persona import Persona
from app.domain.exceptions import NotFoundError


class SetDefaultPersonaUseCase:
    def __init__(self, repo: PersonasRepository) -> None:
        self.repo = repo

    def execute(self, *, persona_id: int, user_id: str) -> Persona:
        try:
            return self.repo.set_default(persona_id, user_id=user_id)
        except NotFoundError:
            raise

File: apps/web/src/application/use-cases/set-default-persona.ts
/**
 * SetDefaultPersonaUseCase — mark a persona as the user's default.
 *
 * Backend ensures only one persona per user holds `is_default=true`;
 * any previously-default persona is auto-unmarked.
 */

import type { PersonasApiPort } from "@/application/ports/personas-api.port";
import type { Persona } from "@/domain/entities/persona";

export class SetDefaultPersonaUseCase {
  constructor(private readonly api: PersonasApiPort) {}

  execute(id: number): Promise<Persona> {
    return this.api.setDefault(id);
  }
}
